File: Chess/src/Pieces/Queen.cpp
#include <cstdlib>
#include "Queen.hpp"
#include "Bishop.hpp"
#include "Rook.hpp"

bool Queen::canMove(sf::Vector2f position) {
	sf::Vector2f current = this->getBoardPosition();
	return Bishop::canMove(current, position) || Rook::canMove(current, position);
}

int Queen::getTextureIndex() {
	return 1;
}

File: Chess.NET/Pieces/Queen.cs
using System;
using System.Collections.Generic;
using System.Linq;
using SFML.System;

namespace Chess.Pieces {
	public class Queen : Piece {
		public Queen(Vector2f position, Team team) : base(position, team) { }

		public override int TextureIndex => 1;

		public override bool CanMove(List<Piece> board, Vector2f to) {
			return Bishop.CanMove(this, board, to) || Rook.CanMove(this, board, to);
		}
	}
}
